src/formatters/default_formatter.erl

```diff
@@ -646,8 +646,13 @@ lay_no_comments(Node, Ctxt) ->
             Ctxt1 = reset_prec(Ctxt),
             D1 = beside(beside(lay_text_float("#"), lay(erl_syntax:record_expr_type(Node), Ctxt1)),
                         text("{")),
-            D2 = lay_fields(D1, erl_syntax:record_expr_fields(Node), Ctxt1, fun lay/2),
             Arg = erl_syntax:record_expr_argument(Node),
+            D2 = case Arg of
```

Comment:
Why are you branching on `Arg` being `none`?
I mean… the request of not adding space between `{` and `}` should apply even if you write `AVar#a_record{}`.
No idea why anybody would do that… but it's perfectly valid Erlang syntax.
Maybe it would be better to do `case erl_syntax:record_expr_fields(Node) of [] -> …; Fields -> … end`, instead.